### DISCUSSION TITLE
Using a param in serverless.yml service:

### DISCUSSION THREAD

**Discussion by silvios** [serverless/serverless#11355] [General]
Hey,

I have a project where I run different serverless configurations based on param values. This project was built using serverless v2. I am looking to upgrade to serverless v3.

I am getting this error: 
```
Serverless Error ----------------------------------------

  Cannot resolve serverless.yml: "service" property is not accessible (configured behind variables which cannot be resolved at this stage)

```

In my serverless.yml this is not the full config but just to show what it looks like. Its used to build many services which are similar but may change slightly

```
service: ${opt:sitename}-some-service-name

resources:
 - ${file(resources/${opt:sitename}/${opt:environment}-${opt:sitename}.yml)}
```
 
 

When attempting to upgrade to Serverless v3 I cannot seem to figure out how to allow params to appear in the service name. Does anyone have a solution or could help point me in the right direction?

**Comment by mnapoli** [comment]
Hi @silvios, since v3 this isn't possible to use a variable in the service name unfortunately.

**Reply by silvios** [reply]
Thanks for the reply @mnapoli if I have a project which has created many services which each have individual resources from the same/similar templates (just properties would have changed and is unique to each service)

Do you have any suggestions on how we can continue to maintain and build out our infrastructure while being able to upgrade to Serverless v3?

**Reply by bryanhorstmann** [reply]
Hi @mnapoli , I've just spent a few hours trying to debug this. If it is no longer supported, could the documentation be updated?

I was attempting to use SSM for service name following this guide: https://www.serverless.com/framework/docs/providers/aws/guide/variables#reference-variables-using-the-ssm-parameter-store

**Reply by mnapoli** [reply]
@bryanhorstmann that's a good point, a PR would likely be welcome (I am no longer a maintainer of Serverless Framework unfortunately).
